src/main/java/dev/sigstore/fulcio/client/FulcioVerifier.java

```diff
@@ -59,42 +65,76 @@ public static FulcioVerifier newFulcioVerifier(byte[] fulcioRoot, byte @Nullable
   }
 
-  public void verifySct(SigningCertificate sc) throws FulcioVerificationException {
-
-    SignedCertificateTimestamp sct =
-        sc.getSct()
-            .orElseThrow(() -> new FulcioVerificationException("No SCT was found to verify"));
-    if (ctLogInfo == null) {
+  /**
+   * Verify that an SCT associated with a Singing Certificate is valid and signed by the configured
+   * CT-log public key.
+   *
+   * @param signingCertificate containing the SCT metadata to verify
+   * @throws FulcioVerificationException if verification fails for any reason
+   */
+  public void verifySct(SigningCertificate signingCertificate) throws FulcioVerificationException {
+    if (ctLogInfo == null || ctVerifier == null) {
       throw new FulcioVerificationException("No ct-log public key was provided to verifier");
     }
 
-    // leaf certificate are guaranteed to be X509Certificates if they were built via
-    // a client request.
-    if (!(sc.getLeafCertificate() instanceof X509Certificate)) {
-      throw new RuntimeException(
-          "Encountered non X509 Certificate when verifying SCT. Leaf certificate is "
-              + sc.getLeafCertificate().getClass());
-    }
-    CertificateEntry ce;
-
-    try {
-      ce = CertificateEntry.createForX509Certificate((X509Certificate) sc.getLeafCertificate());
-    } catch (CertificateEncodingException cee) {
-      throw new FulcioVerificationException("Leaf certificate could not be parsed", cee);
-    }
-
-    VerifiedSCT.Status status = ctLogInfo.verifySingleSCT(sct, ce);
-    if (status != VerifiedSCT.Status.VALID) {
-      throw new FulcioVerificationException(
-          "SCT could not be verified because " + status.toString());
+    if (signingCertificate.getDetachedSct().isPresent()) {
+      CertificateEntry ce;
+      try {
+        ce = CertificateEntry.createForX509Certificate(signingCertificate.getLeafCertificate());
+      } catch (CertificateEncodingException cee) {
+        throw new FulcioVerificationException("Leaf certificate could not be parsed", cee);
+      }
+
+      var status = ctLogInfo.verifySingleSCT(signingCertificate.getDetachedSct().get(), ce);
+      if (status != VerifiedSCT.Status.VALID) {
+        throw new FulcioVerificationException(
+            "SCT could not be verified because " + status.toString());
+      }
+    } else if (signingCertificate.hasEmbeddedSct()) {
+      var certs = signingCertificate.getCertificates();
+      CTVerificationResult result;
+      try {
+        result = ctVerifier.verifySignedCertificateTimestamps(certs, null, null);
+      } catch (CertificateEncodingException cee) {
+        throw new FulcioVerificationException(
+            "Certificates could not be parsed during sct verification");
+      }
+      int valid = result.getValidSCTs().size();
+      int invalid = result.getInvalidSCTs().size();
+      if (valid != 1 || invalid != 0) {
```

Comment:
i wouldn't hardcode valid to 1. at some point, we may start including multiple SCTs from multiple logs in certificates. you just need to iterate through all SCTs and verify each

Reply:
Makes sense.

But, on the topic of multiple logs -- where would these logs be? Will the logs public key be distributed via tuf root? I'm not sure how this works.

Reply:
Unclear at this point, maybe delegated keys